File: ui/Components/EntryExit/Dtos/EntryExitDto.ts

import { EntryExitEnum } from "../../../Enums/EntryExitEnum";
import { initialLocationDto, LocationDto } from "../../../Location/Dtos/locationDto";

export interface EntryExitDto {
    id: number;
    locationDto: LocationDto;
    barcodeReadEnum: EntryExitEnum | null;
    biometricEnum: EntryExitEnum | null;
    adminApproveEnum:EntryExitEnum | null;
    userId: number;
    roleId: number;
    deviceId: number;
    status: 'idle' | 'loading' | 'succeeded' | 'failed';
    biometricIconVisible: boolean
    qrCodeIconVisible: boolean
    isUserCompleteShift:boolean
}
export const initialEntryExitDto: EntryExitDto = {
    id: 0,
    locationDto: initialLocationDto,
    barcodeReadEnum: null,
    biometricEnum: null,
    userId: 0,
    roleId: 0,
    deviceId: 0,
    status: "idle",
    biometricIconVisible: true,
    qrCodeIconVisible: true,
    isUserCompleteShift: false,
    adminApproveEnum: null
}

File: api/Bussiness/Services/EntryExitService/Dtos/EntryExitDto.cs
using Bussiness.Helper.Dtos;
using Bussiness.Helper.Enums;
namespace Bussiness.Services.EntryExitService.Dtos
{
    public class EntryExitDto
    {
        public int Id { get; set; }
        public LocationDto LocationDto { get; set; }     // Cihaz kullanıcısının koordinatları
        public EntryExitEnum? BarcodeReadEnum { get; set; }
        public EntryExitEnum? BiometricEnum { get; set; }
        public EntryExitEnum? AdminApproveEnum { get; set; }
        public int UserId { get; set; }
        public int RoleId { get; set; }
        public int DeviceId { get; set; }
        public bool IsUserCompleteShift { get; set; }
    }
}
